libglusterfs/src/dict.c

```diff
@@ -30,6 +30,8 @@ struct dict_cmp {
     gf_boolean_t (*value_ignore)(char *k);
 };
 
+static glusterfs_ctx_t *global_ctx = NULL;
```

Comment:
`THIS->ctx` is not specific to dicts. If you want to use it globally, you should put it in the right place. Probably globals.c would be a good place for it.